### PR TITLE
feat: updated GitHub star count number format

### LINKED ISSUE #1320: ✨ [Feature request]: Show GitHub Star Count in formatted way
### Is your feature request related to a problem? Please describe.

Right now the Github Star Count is unformatted.. like 1091.. we can show 1.1K instead.

### Describe the solution you'd like.

May be we can use a better library to format all the numbers with K, M etc,

### Describe alternatives you've considered.

None

### Screenshot / Screenshare

image

### Additional context

_No response_

### Record

- [X] I have checked the existing [issues](https://github.com/reactplay/react-play/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/reactplay/react-play/blob/main/CONTRIBUTING.md)
- [X] I want to work on this issue
- [X] I am in Hacktoberfest

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by FreemRL**
# Description

I changed the GitHub star count number format to show numbers in a more efficient way --> e.g. `1.1K`, `1.1M` etc.

Fixes #1320 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I used the development server to test functionality. `yarn start`.

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output

Before:

before

After:

after

### CODE REVIEW

**Comment by atapas** [comment]
> https://deploy-preview-1321--reactplayio.netlify.app/

Hey @FreemRL I do not see the change reflecting in the preview build above.

**Comment by FreemRL** [comment]
Okay I forgot to change it at `DefaultBanner.jsx` as well, sorry for that. Does it work now? @atapas

**Comment by FreemRL** [comment]
I can't deploy my changes at Netlify:

image

image

I think therefor it can't be shown in the deploy preview. Any idea how I can fix the deploy on Netlify? @atapas

**Comment by FreemRL** [comment]
UPDATE:

I found a way to get the initialization done but now the build part is the next problem:
image

image

This results in the skip of deployment:
image

**Comment by Sarhan41** [comment]
Hey , i had checked this issue, where i go to api endpoint on browser, it till says 1093 instead of real count. 
so it is issue with the api or there backend. i don't see any issue in the api request or fetchin it , data in the json file of that api is wrong.
Screenshot 2023-10-11 071538
you can go here and check [GITHUBAPI](https://api.github.com/repos/reactplay/react-play).

**Comment by FreemRL** [comment]
@atapas I got around this message:

image

However building still fails so I can't deploy my changes to netlify which is the reason why you don't see my changes at https://deploy-preview-1321--reactplayio.netlify.app/

I built the app locally and served it. This is the result I get:
image

As you can see my changes work just fine.

But also if I build the app locally I get this warning message:
image

So to sum it up: My changes work but I can't deploy them to netlify. What should I do now?

**Comment by atapas** [comment]
@FreemRL I have kicked off the build.. will check in a while.

File: package.json [hunk 1]
```diff
@@ -125,6 +125,7 @@
     ]
   },
   "devDependencies": {
+    "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
```
Comment (atapas) [inline package.json:128]: @FreemRL Why do you need this package?

Comment (FreemRL) [reply package.json:128]: I can remove it if you want, I added it to fix this warning:

image

**Review by atapas — APPROVED** [review]
Changes looking good.
